styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesDockerRunner.java

```diff
@@ -341,21 +343,23 @@ static EnvVar envVar(String name, String value) {
   private static List<EnvVar> buildEnv(WorkflowInstance workflowInstance,
                                        RunSpec runSpec,
                                        String styxEnvironment) {
-    return Arrays.asList(
-        envVar(COMPONENT_ID,    workflowInstance.workflowId().componentId()),
-        envVar(WORKFLOW_ID,     workflowInstance.workflowId().id()),
-        envVar(PARAMETER,       workflowInstance.parameter()),
-        envVar(COMMIT_SHA,      runSpec.commitSha().orElse("")),
-        envVar(SERVICE_ACCOUNT, runSpec.serviceAccount().orElse("")),
-        envVar(DOCKER_ARGS,     String.join(" ", runSpec.args())),
-        envVar(DOCKER_IMAGE,    runSpec.imageName()),
-        envVar(EXECUTION_ID,    runSpec.executionId()),
-        envVar(TERMINATION_LOG, "/dev/termination-log"),
-        envVar(TRIGGER_ID,      runSpec.trigger().map(TriggerUtil::triggerId).orElse(null)),
-        envVar(TRIGGER_TYPE,    runSpec.trigger().map(TriggerUtil::triggerType).orElse(null)),
-        envVar(ENVIRONMENT,     styxEnvironment),
-        envVar(LOGGING,         "structured")
-    );
+    final Map<String, String> env = new HashMap<>(runSpec.env());
```

Comment:
Maybe comment here it is done on purpose to prevent user provided envs overwriting those system ones?